### ISSUE TITLE
duplicate drifts detected by ADWIN

### ISSUE THREAD

**Issue by remiadon** [online-ml/river#574]
## Versions
**river version**: 0.0.7
**Python version**: Python 3.9.2
**Operating system**: macOS Sierra 10.12.6 (16G29)

## Describe the bug
I tried to run the following workflow:
 - generate abrupt drifts for a variable, at predefined intervals (by forcing the average value to change)
 - detect drift with river.drfit.ADWIN

ADWIN detects drifts more than once (usually twice, even more when setting parameter λ to a very low value)

There is a very similar workflow in [river's doc](https://riverml.xyz/dev/examples/concept-drift-detection/), that also clearly shows there is a problem.

## Steps/code to reproduce
copied from [the doc](https://riverml.xyz/dev/examples/concept-drift-detection/) (at the bottom of the page)
```python
import numpy as np
import matplotlib.pyplot as plt
from matplotlib import gridspec

from river import drift

# Generate data for 3 distributions
random_state = np.random.RandomState(seed=42)
dist_a = random_state.normal(0.8, 0.05, 1000)
dist_b = random_state.normal(0.4, 0.02, 1000)
dist_c = random_state.normal(0.6, 0.1, 1000)

# Concatenate data to simulate a data stream with 2 drifts
stream = np.concatenate((dist_a, dist_b, dist_c))

# Auxiliary function to plot the data
def plot_data(dist_a, dist_b, dist_c, drifts=None):
    fig = plt.figure(figsize=(7,3), tight_layout=True)
    gs = gridspec.GridSpec(1, 2, width_ratios=[3, 1])
    ax1, ax2 = plt.subplot(gs[0]), plt.subplot(gs[1])
    ax1.grid()
    ax1.plot(stream, label='Stream')
    ax2.grid(axis='y')
    ax2.hist(dist_a, label=r'$dist_a$')
    ax2.hist(dist_b, label=r'$dist_b$')
    ax2.hist(dist_c, label=r'$dist_c$')
    if drifts is not None:
        for drift_detected in drifts:
            ax1.axvline(drift_detected, color='red')
    plt.show()

drift_detector = drift.ADWIN()
drifts = []

for i, val in enumerate(stream):
    drift_detector.update(val)   # Data is processed one sample at a time
    if drift_detector.change_detected:
        # The drift detector indicates after each sample if there is a drift in the data
        print(f'Change detected at index {i}')
        drifts.append(i)
        drift_detector.reset()   # As a best practice, we reset the detector

plot_data(dist_a, dist_b, dist_c, drifts)
```
<img width="686" alt="Screen Shot 2021-04-30 at 18 56 47" src="https://user-images.githubusercontent.com/2931080/116728273-f9e3e880-a9e5-11eb-8c89-36416942d36a.png">

Now here is the output if I replace 
```python
drift_detector.reset()   # As a best practice, we reset the detector
```
by 
```python
drift_detector = drift.ADWIN()  # brand new instance instead of calling .reset() 
```
<img width="686" alt="Screen Shot 2021-04-30 at 19 01 00" src="https://user-images.githubusercontent.com/2931080/116728632-6f4fb900-a9e6-11eb-97df-c39eff935417.png">

## How to solve it ?
The [same issue](https://github.com/scikit-multiflow/scikit-multiflow/issues/99) was raised in scikit-multiflow

in the current version of river the `reset` method is defined [here](https://github.com/online-ml/river/blob/1d269ae8ab6eb1c3973b308dcba394532b7998ff/river/drift/adwin_c.pyx#L42), whereas `river.base.DriftDector` does not implement a `reset` method.
My guess is that the there is an issue in which __init__ function is called when actually calling `reset`

I can investigate and submit a PR

NB. If resetting the detector is a "good practice" as mentioned in the doc, why not providing a `reset=True` default parameter for the `ADWIN` class ?

**Comment by MaxHalford** [comment]
@hoanganhngo610 maybe you can be of assistance here?

**Comment by jacobmontiel** [comment]
I am taking this :-)

This is indeed a bug. The reset method was lost during the Cython refactoring. The default reset method in the base class does not reset the ADWIN helper. The fix is in #576.

Thanks @remiadon for the quick catch on this and the great MWE 🙌 , it made the fix easier.
